build/cross_compile.sh

```diff
@@ -0,0 +1,168 @@
+PROGRAM=`basename $0`
+
+BUILDDIR=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
+CORE=`grep -c ^processor /proc/cpuinfo`
+CMAKE=cmake
+OS_HOST=`grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
+
+if [[ $OS_HOST == "centos" ]] || [[ $OS_HOST == "rhel" ]]; then
+    CMAKE=cmake3
+    if [[ ! -x "$(command -v $CMAKE)" ]]; then
+        echo "$CMAKE is not installed, please install it"
+        exit 1
+    fi
+fi
+
+clean=0
+jcore=$CORE
+
+while [ $# -gt 0 ]; do
+        case $1 in
+                -help )
+                        usage_and_exit 0
+                        ;;
+                -aarch )
+                        shift
+                        AARCH=$1
+                        ;;
+                -sysroot )
+                        shift
+                        SYSROOT=$1
+                        ;;
+                -clean | clean )
+                        clean=1
+                        ;;
+		-compiler )
+			shift
+			COMPILER=$1
+			;;
+                 --* | -* )
+                        error "Unregognized option: $1"
+                        ;;
+                 * )
+                        error "Unregognized option: $1"
+                        ;;
+        esac
+        shift
+done
+
+edge_dir_aarch64="Edge_aarch64"
+edge_dir_aarch32="Edge_aarch32"
+toolchain="toolchain-edge.cmake"
+
+here=$PWD
+cd $BUILDDIR
+
+if [[ $clean == 1 ]]; then
+    echo $PWD
+    echo "/bin/rm -rf $edge_dir_aarch64 $edge_dir_aarch32"
+    /bin/rm -rf $edge_dir_aarch64 $edge_dir_aarch32 
+    exit 0
+fi
+ 
+# Sanity Check
+if [ -z $AARCH ] || [ -z $SYSROOT ] ; then
+    error "Please provide the required options 'aarch', 'sysroot'"
+fi
+
+if [ ! -d $SYSROOT ]; then
+    error "SYSROOT is not accessible"
+fi
+
+if [ $AARCH == "aarch64" ]; then
+    mkdir -p $edge_dir_aarch64 
+    cd $edge_dir_aarch64
+elif [ $AARCH == "aarch32" ]; then
+    mkdir -p $edge_dir_aarch32
+    cd $edge_dir_aarch32
+    AARCH="arm"
+else
+    error "$AARCH not exist"
+fi
+
+if [ -z $COMPILER ]; then
+    echo "INFO: Path to gcc cross compiler is not specified, will be searched in default paths in host"
+    if [ $AARCH == "aarch64" ]; then
+      target_abi="linux-gnu"
+      c_compiler="aarch64-${target_abi}-gcc"
+      cxx_compiler="aarch64-${target_abi}-g++"
+    else
+      target_abi="linux-gnueabihf"
+      c_compiler="arm-${target_abi}-gcc"
+      cxx_compiler="arm-${target_abi}-g++"
+    fi
+else
+    c_compiler=$COMPILER
+    dir=$(dirname "${COMPILER}")
+    if [ $AARCH == "aarch64" ]; then
+      cxx_compiler=${dir}/aarch64-linux-gnu-g++
+    else
+      cxx_compiler=${dir}/arm-linux-gnueabihf-g++
+    fi
+fi
+
+if [ ! -f ${SYSROOT}/etc/os-release ]; then
+    #Use case for Petalinux Sysroot
+    echo "INFO: OS Flavor of SYSROOT cannot be determined, generating RPM packages of XRT"
+    OS_TARGET="CentOS"
+    echo "INFO: Cross Compiling XRT for $AARCH" 
+else
+    OS_TARGET=`grep '^ID=' ${SYSROOT}/etc/os-release | awk -F= '{print $2}' | tr -d '"'`
```

Comment:
upper-casing of Ubuntu and Debian is needed as it will be useful in cpackLin.cmake file where it expects value to be Ubuntu|Debian. Yeah and I missed to handle Debian case and will update the code accordingly.
Sourcing can be done but I followed the similar style of getting this value like in other places of XRT